File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let res = 0
    let min = Number.MAX_VALUE
    
    prices.forEach(p => {
        min = Math.min(min, p)
        res = Math.max(p - min, res)
    })
    
    return res
};

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int result = 0;
        int currentPrice = Integer.MAX_VALUE;
        
        for(int p : prices) {
            result = Math.max(p - currentPrice, result);
            currentPrice = Math.min(currentPrice, p);
        }
        
        return result;
    }
}